quickwit-metastore/src/metastore/mod.rs

```diff
@@ -69,10 +69,13 @@ pub trait Metastore: Send + Sync + 'static {
     /// Checks whether the metastore is available.
     async fn check_connectivity(&self) -> anyhow::Result<()>;
 
-    /// Checks whether the given index is in this metastore.
-    async fn check_index_available(&self, index_id: &str) -> anyhow::Result<()> {
```

Comment:
This `check_index_available` can make you believe that an existing index is not present when there is a network error.